# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediacontrol/MediaControlService.cpp
# Commit: 3404fd6725f1
# Full Hash: 3404fd6725f15d1253a7f76e3937185bd353f263
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part2: Switch Localization class to use localization-ffi. r=emilio,nika
#   
#   Depends on D104788
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104789
# ==============================================================================

diff -r c9cfbba15d90 -r 3404fd6725f1 dom/media/mediacontrol/MediaControlService.cpp
--- a/dom/media/mediacontrol/MediaControlService.cpp	Mon Aug 02 09:54:07 2021 +0000
+++ b/dom/media/mediacontrol/MediaControlService.cpp	Mon Aug 02 09:54:07 2021 +0000
@@ -114,18 +114,15 @@
   mControllerManager = MakeUnique<ControllerManager>(this);
 
   // Initialize the fallback title
-  nsCOMPtr<nsIGlobalObject> global =
-      xpc::NativeGlobal(xpc::PrivilegedJunkScope());
-  RefPtr<Localization> l10n = Localization::Create(global, true, {});
-  l10n->AddResourceId(u"branding/brand.ftl"_ns);
-  l10n->AddResourceId(u"dom/media.ftl"_ns);
+  nsTArray<nsCString> resIds{
+      "branding/brand.ftl"_ns,
+      "dom/media.ftl"_ns,
+  };
+  RefPtr<Localization> l10n = Localization::Create(resIds, true);
   {
-    AutoSafeJSContext cx;
-
     nsAutoCString translation;
-    ErrorResult rv;
-    l10n->FormatValueSync(cx, "mediastatus-fallback-title"_ns, {}, translation,
-                          rv);
+    IgnoredErrorResult rv;
+    l10n->FormatValueSync("mediastatus-fallback-title"_ns, {}, translation, rv);
     if (!rv.Failed()) {
       mFallbackTitle = NS_ConvertUTF8toUTF16(translation);
     }